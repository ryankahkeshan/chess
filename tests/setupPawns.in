setup 
+ K e1
+ k e8
+ P a1
+ P a2
+ P a3
+ P b5
+ P a7
+ P a8
done
- a1
done
- a8
done
