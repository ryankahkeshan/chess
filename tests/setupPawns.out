8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 P _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 P_ _ _ _
1 P _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 P _ _ _ 
2 P_ _ _ _
1 P _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _P_ _ _ 
4  _ _ _ _
3 P _ _ _ 
2 P_ _ _ _
1 P _ K _ 

  abcdefgh

8  _ _k_ _
7 P _ _ _ 
6  _ _ _ _
5 _P_ _ _ 
4  _ _ _ _
3 P _ _ _ 
2 P_ _ _ _
1 P _ K _ 

  abcdefgh

8 P_ _k_ _
7 P _ _ _ 
6  _ _ _ _
5 _P_ _ _ 
4  _ _ _ _
3 P _ _ _ 
2 P_ _ _ _
1 P _ K _ 

  abcdefgh

Invalid Setup: 
  - Each player must have exactly one king
  - There should be no pawns on the first or last row
8 P_ _k_ _
7 P _ _ _ 
6  _ _ _ _
5 _P_ _ _ 
4  _ _ _ _
3 P _ _ _ 
2 P_ _ _ _
1 _ _ K _ 

  abcdefgh

Invalid Setup: 
  - Each player must have exactly one king
  - There should be no pawns on the first or last row
8  _ _k_ _
7 P _ _ _ 
6  _ _ _ _
5 _P_ _ _ 
4  _ _ _ _
3 P _ _ _ 
2 P_ _ _ _
1 _ _ K _ 

  abcdefgh

Final Score:
White: 0
Black: 0
